distribution/tools/plugin-cli/src/test/java/org/opensearch/plugins/InstallPluginCommandTests.java

```diff
@@ -256,29 +256,52 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
     }
 
     /** creates a plugin .zip and returns the url for testing */
-    static String createPluginUrl(String name, Path structure, String... additionalProps) throws IOException {
-        return createPlugin(name, structure, additionalProps).toUri().toURL().toString();
-    }
-
-    static void writePlugin(String name, Path structure, String... additionalProps) throws IOException {
-        String[] properties = Stream.concat(
-            Stream.of(
-                "description",
-                "fake desc",
-                "name",
-                name,
-                "version",
-                "1.0",
-                "opensearch.version",
-                Version.CURRENT.toString(),
-                "java.version",
-                System.getProperty("java.specification.version"),
-                "classname",
-                "FakePlugin"
-            ),
-            Arrays.stream(additionalProps)
-        ).toArray(String[]::new);
-        PluginTestUtil.writePluginProperties(structure, properties);
+    static String createPluginUrl(String name, boolean withCustomFolder, Path structure, String... additionalProps) throws IOException {
+        return createPlugin(name, withCustomFolder, structure, additionalProps).toUri().toURL().toString();
+    }
+
+    static void writePlugin(String name, boolean withCustomFolder, Path structure, String... additionalProps) throws IOException {
+        if (withCustomFolder) {
```

Comment:
Pass in the actual folder name (or null) instead of a boolean, and set it instead of doing this `if`. Avoids duplicating this.

Reply:
Updated this.